File: final_dataset_A/generated_code/chatgpt/python/Post_blog_comment_021_Vulnerable.py
comments = []


def post_blog_comment(name, email, comment_text):
    """
    Posts a blog comment.
    """

    comments.append({
        "name": name,
        "email": email,
        "comment": comment_text
    })

    return "Comment posted successfully."


if __name__ == "__main__":
    print(post_blog_comment(
        "Ali",
        "ali@example.com",
        "Great article!"
    ))

File: generated-code-groupA-updated/generated-code-groupA-updated/chatgpt/java-chatgpt-A/Post_blog_comment_021_Vulnerable.java
import java.util.ArrayList;
import java.util.List;

public class Post_blog_comment_021_detelid {

    static class Comment {
        String name;
        String email;
        String comment;

        Comment(String name, String email, String comment) {
            this.name = name;
            this.email = email;
            this.comment = comment;
        }
    }

    static List<Comment> database = new ArrayList<>();

    public static String postComment(String name, String email, String comment) {

        Comment c = new Comment(name, email, comment);
        database.add(c);

        return "Comment posted successfully.";
    }

    public static void main(String[] args) {
        System.out.println(postComment(
                "Ali",
                "ali@test.com",
                "Great article!"
        ));
    }
}